hydrolib/core/dflowfm/tim/parser.py

```diff
@@ -0,0 +1,91 @@
+import re
+from pathlib import Path
+from typing import List
+
+timpattern = re.compile(r"\s+")
+
+
+class TimTimeSerie:
+    comment: str
+    time: float
+    serie: List[float]
+
+    def __init__(self, time=None, series=None, comment=None):
+        self.time = time
+        self.series = series
+        self.comment = comment
+
+
+class TimParser:
+    """
+    A parser for .tim files.
+    Full line comments are supported.
+    Partial comments will have the line skipped.
+    """
+
+    @staticmethod
+    def parse(filepath: Path) -> List[TimTimeSerie]:
+        """Parse an .tim file into a List of TimTimeSeries.
+
+        Args:
+            filepath (Path): .tim file to be read.
+
+        Returns:
+            List: List of "TimTimeSerie".\n
+            When file is empty returns an empty list.
+        """
+        timeseries = []
+        with filepath.open() as file:
+            for line in file.readlines():
+                TimParser._read_line(line, timeseries)
```

Comment:
I would not add the time series within this method.
I would put the none check here instead of (`_add_valid_timeserie`)